examples/bpi-monolith-test-suite/bpi/components/zkcircuits/circuitlib/orderCircuit.go

```diff
@@ -0,0 +1,183 @@
+package circuitlib
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"zkcircuit/json"
+
+	"github.com/consensys/gnark-crypto/ecc"
+	"github.com/consensys/gnark/backend"
+	"github.com/consensys/gnark/backend/groth16"
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/hash/mimc"
+)
+
+type OrderCircuit struct {
+	I  Input
+	PI PrivateInput
+}
+
+type Input struct {
+	AgreementStateCommitment frontend.Variable `gnark:",public"`
+	StateObjectCommitment    frontend.Variable `gnark:",public"`
+	CalculatedAgreementRoot  frontend.Variable `gnark:",public"`
+}
+
+type PrivateInput struct {
+	O                  Order
+	OrderRoot          frontend.Variable `gnark:",secret"`
+	OrderSalt          frontend.Variable `gnark:",secret"`
+	BuyerPK            frontend.Variable `gnark:",secret"`
+	AgreementStateRoot frontend.Variable `gnark:",secret"`
+	AgreementStateSalt frontend.Variable `gnark:",secret"`
+}
+
+type Order struct {
+	ProductId frontend.Variable `gnark:",secret"`
+	BuyerSig  frontend.Variable `gnark:",secret"`
+}
+
+var verifyingKey string
+
+func (circuit *OrderCircuit) Define(api frontend.API) error {
+	mimc, _ := mimc.NewMiMC(api)
+
+	// Asserts Input AgreementStateCommitment and Derived AgreementStateCommitment are equal
+	sum := api.Add(circuit.PI.AgreementStateRoot, circuit.PI.AgreementStateSalt)
+	mimc.Write(sum)
+	api.AssertIsEqual(circuit.I.AgreementStateCommitment, mimc.Sum())
+
+	// Asserts Input StateObjectCommitment and Derived StateObjectCommitment are equal
+	mimc.Reset()
+	sum = api.Add(circuit.PI.OrderRoot, circuit.PI.OrderSalt)
+	mimc.Write(sum)
+	api.AssertIsEqual(circuit.I.StateObjectCommitment, api.Add(circuit.PI.OrderRoot, circuit.PI.OrderSalt))
+
+	//Asserts CalculatedAgreementRoot and PrivateInput AgreementStateRoot are equal
+	api.AssertIsEqual(circuit.I.CalculatedAgreementRoot, circuit.PI.AgreementStateRoot)
+
+	//Verifies Signature is correct - mimics mock fashion
+	api.AssertIsEqual(circuit.PI.O.BuyerSig, api.Add(circuit.PI.BuyerPK, circuit.PI.O.ProductId))
+
+	return nil
+}
+
+func GenerateProof(p json.Input) (groth16.Proof, string) {
```

Comment:
can we just return (groth16.Proof, error) instead of string?

Reply:
error returned instead of string